lib/web/integrations.go

```diff
@@ -197,6 +201,122 @@ func (h *Handler) integrationsGet(w http.ResponseWriter, r *http.Request, p http
+
+	summary, err := collectAWSOIDCAutoDiscoverStats(r.Context(), ig, clt.DiscoveryConfigClient())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return summary, nil
+}
+
+func collectAWSOIDCAutoDiscoverStats(
+	ctx context.Context,
+	integration types.Integration,
+	clt interface {
+		ListDiscoveryConfigs(ctx context.Context, pageSize int, nextToken string) ([]*discoveryconfig.DiscoveryConfig, string, error)
+	},
+) (ui.IntegrationWithSummary, error) {
+	var ret ui.IntegrationWithSummary
+
+	uiIg, err := ui.MakeIntegration(integration)
+	if err != nil {
+		return ret, err
+	}
+	ret.Integration = uiIg
+
+	var nextPage string
+	for {
+		discoveryConfigs, nextToken, err := clt.ListDiscoveryConfigs(ctx, 0, nextPage)
+		if err != nil {
+			return ret, trace.Wrap(err)
+		}
+		for _, dc := range discoveryConfigs {
+			discoveredResources, ok := dc.Status.IntegrationDiscoveredResources[integration.GetName()]
+			if !ok {
+				continue
+			}
+
+			if matchers := matchersWithIntegration(dc, types.AWSMatcherEC2, integration.GetName()); matchers != 0 {
+				ret.AWSEC2.RulesCount = ret.AWSEC2.RulesCount + matchers
+				ret.AWSEC2.DiscoverLastSync = lastSync(ret.AWSEC2.DiscoverLastSync, dc.Status.LastSyncTime)
+				ret.AWSEC2.ResourcesFound = ret.AWSEC2.ResourcesFound + int(discoveredResources.AwsEc2.Found)
+				ret.AWSEC2.ResourcesEnrollmentSuccess = ret.AWSEC2.ResourcesEnrollmentSuccess + int(discoveredResources.AwsEc2.Enrolled)
+				ret.AWSEC2.ResourcesEnrollmentFailed = ret.AWSEC2.ResourcesEnrollmentFailed + int(discoveredResources.AwsEc2.Failed)
+			}
+
+			if matchers := matchersWithIntegration(dc, types.AWSMatcherRDS, integration.GetName()); matchers != 0 {
+				ret.AWSRDS.RulesCount = ret.AWSRDS.RulesCount + matchers
+				ret.AWSRDS.DiscoverLastSync = lastSync(ret.AWSRDS.DiscoverLastSync, dc.Status.LastSyncTime)
+				ret.AWSRDS.ResourcesFound = ret.AWSRDS.ResourcesFound + int(discoveredResources.AwsRds.Found)
+				ret.AWSRDS.ResourcesEnrollmentSuccess = ret.AWSRDS.ResourcesEnrollmentSuccess + int(discoveredResources.AwsRds.Enrolled)
+				ret.AWSRDS.ResourcesEnrollmentFailed = ret.AWSRDS.ResourcesEnrollmentFailed + int(discoveredResources.AwsRds.Failed)
+			}
+
+			if matchers := matchersWithIntegration(dc, types.AWSMatcherEKS, integration.GetName()); matchers != 0 {
+				ret.AWSEKS.RulesCount = ret.AWSEKS.RulesCount + matchers
+				ret.AWSEKS.DiscoverLastSync = lastSync(ret.AWSEKS.DiscoverLastSync, dc.Status.LastSyncTime)
+				ret.AWSEKS.ResourcesFound = ret.AWSEKS.ResourcesFound + int(discoveredResources.AwsEks.Found)
+				ret.AWSEKS.ResourcesEnrollmentSuccess = ret.AWSEKS.ResourcesEnrollmentSuccess + int(discoveredResources.AwsEks.Enrolled)
+				ret.AWSEKS.ResourcesEnrollmentFailed = ret.AWSEKS.ResourcesEnrollmentFailed + int(discoveredResources.AwsEks.Failed)
+			}
+		}
+
+		if nextToken == "" {
+			break
+		}
+		nextPage = nextToken
+	}
+
+	// TODO(marco): add total number of ECS Database Services.
+	ret.AWSRDS.ECSDatabaseServiceCount = 0
```

Comment:
Should we include the ECS count if it's always 0? Or will this be coming soon? (Just thinking it might get a bug report if it's left as 0 in a release)

Reply:
This is coming soon.
As soon as the following PR lands, I'll change this to include the count:
https://github.com/gravitational/teleport/pull/49352